components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -231,29 +258,11 @@ func dbOIDCClientConfigsToAPI(configs []db.OIDCClientConfig, decryptor db.Decryp
 	return results, nil
 }
 
-func papiConfigToIAM(c *v1.OIDCClientConfig) *iam.OIDCClientConfig {
-	return &iam.OIDCClientConfig{
-		OrganizationId: c.GetOrganizationId(),
-		OidcConfig: &iam.OIDCConfig{
-			Issuer: c.OidcConfig.GetIssuer(),
-		},
-		Oauth2Config: &iam.OAuth2Config{
-			ClientId:     c.GetOauth2Config().GetClientId(),
-			ClientSecret: c.GetOauth2Config().GetClientSecret(),
-		},
-	}
-}
-func iamConfigToPAPI(c *iam.OIDCClientConfig) *v1.OIDCClientConfig {
-	return &v1.OIDCClientConfig{
-		Id:             c.GetId(),
-		OrganizationId: c.GetOrganizationId(),
-		OidcConfig: &v1.OIDCConfig{
-			Issuer: c.OidcConfig.GetIssuer(),
-		},
-		Oauth2Config: &v1.OAuth2Config{
-			ClientId:     c.GetOauth2Config().GetClientId(),
-			ClientSecret: "REDACTED",
-		},
-		CreationTime: c.GetCreationTime(),
+func toDbOIDCSpec(oauth2Config *v1.OAuth2Config, oidcConfig *v1.OIDCConfig) db.OIDCSpec {
+	return db.OIDCSpec{
+		ClientID:     oauth2Config.GetClientId(),
+		ClientSecret: oauth2Config.GetClientSecret(),
+		RedirectURL:  oauth2Config.GetAuthorizationEndpoint(),
+		Scopes:       append([]string{goidc.ScopeOpenID, "profile", "email"}, oauth2Config.GetScopes()...),
```

Comment:
notes (to myself):
1. we need to revisit the shape of payloads, ideally in the coming iteration. flat list like the persistence layer (OIDCSpec) is much easier to handle than the synthetic encapsulation, also mapping forth and back easier. 
2. preset of scopes are ok for google, but we should even remove the "openid" scope from `toDbOIDCSpec` and decide on on another level, based on a) discovery and/or b) user input. for example: this will already break for azure, as they have slightly different scopes required for the `userInfo` endpoint.

I'll create tasks for that.